controllers/managedocs_controller.go

```diff
@@ -255,3 +340,108 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
 
 	return components
 }
+
+func (r *ManagedOCSReconciler) checkUninstallCondition() bool {
+	cmNamespacedName := types.NamespacedName{
+		Name:      r.DeleteConfigMapName,
+		Namespace: r.namespace,
+	}
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(r.ctx, cmNamespacedName, configmap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		r.Log.Error(err, "Unable to get configMap")
+		return false
+	}
+	_, ok := configmap.Labels[r.DeleteConfigMapLabelKey]
+	return ok
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() error {
+
+	r.Log.Info("Deleting subscription")
+	subscription := &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.AddonSubscriptionName,
+			Namespace: r.namespace,
+		},
+	}
+	if err := r.Client.Delete(r.ctx, subscription); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("Unable to delete subscription: %v", err)
+	}
+	r.Log.Info("subscription removed successfully")
+
+	r.Log.Info("Deleting CSV")
+	csvList := &operators.ClusterServiceVersionList{}
+	err := r.Client.List(r.ctx, csvList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list CSVs")
+		return err
+	}
+	var csv operators.ClusterServiceVersion
+	for _, candidate := range csvList.Items {
+		if strings.HasPrefix(candidate.Name, deployerCsvPrefix) {
+			csv = candidate
+			break
+		}
+	}
+	if csv.UID != "" {
```

Comment:
This will create an in-memory copy of every CSV it goes over.
A better way will be to use a pointer:
```suggestion
        var csv &operators.ClusterServiceVersion = nil
	for index := range csvList.Items {
	        candidate := &csvList.Items[index]
		if strings.HasPrefix(candidate.Name, deployerCsvPrefix) {
			csv = candidate
			break
		}
	}
	if csv != nil {
```